programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -1453,6 +1495,101 @@ mod tests {
+                + mem::size_of::<Pubkey>() * 2
+                + mem::size_of::<RcBox<RefCell<&mut u64>>>()
+                + mem::size_of::<u64>()
+                + mem::size_of::<RcBox<RefCell<&mut [u8]>>>()
+                + self.data.len();
+            let mut data = vec![0; size];
+
+            let vm_addr = vm_addr as usize;
+            let key_addr = vm_addr + mem::size_of::<AccountInfo>();
+            let lamports_cell_addr = key_addr + mem::size_of::<Pubkey>();
+            let lamports_addr = lamports_cell_addr + mem::size_of::<RcBox<RefCell<&mut u64>>>();
+            let owner_addr = lamports_addr + mem::size_of::<u64>();
+            let data_cell_addr = owner_addr + mem::size_of::<Pubkey>();
+            let data_addr = data_cell_addr + mem::size_of::<RcBox<RefCell<&mut [u8]>>>();
+
+            let info = AccountInfo {
+                key: unsafe { (key_addr as *const Pubkey).as_ref() }.unwrap(),
+                is_signer: self.is_signer,
+                is_writable: self.is_writable,
+                lamports: unsafe {
+                    Rc::from_raw((lamports_cell_addr + RcBox::<&mut u64>::VALUE_OFFSET) as *const _)
+                },
+                data: unsafe {
+                    Rc::from_raw((data_cell_addr + RcBox::<&mut [u8]>::VALUE_OFFSET) as *const _)
+                },
+                owner: unsafe { (owner_addr as *const Pubkey).as_ref() }.unwrap(),
+                executable: self.executable,
+                rent_epoch: self.rent_epoch,
+            };
+
+            unsafe {
+                ptr::write_unaligned(data.as_mut_ptr().cast(), info);
+                ptr::write_unaligned(
+                    (data.as_mut_ptr() as usize + key_addr - vm_addr) as *mut _,
+                    self.key,
+                );
+                ptr::write_unaligned(
+                    (data.as_mut_ptr() as usize + lamports_cell_addr - vm_addr) as *mut _,
+                    RcBox::new(RefCell::new((lamports_addr as *mut u64).as_mut().unwrap())),
+                );
+                ptr::write_unaligned(
+                    (data.as_mut_ptr() as usize + lamports_addr - vm_addr) as *mut _,
+                    self.lamports,
+                );
+                ptr::write_unaligned(
+                    (data.as_mut_ptr() as usize + owner_addr - vm_addr) as *mut _,
+                    self.owner,
+                );
+                ptr::write_unaligned(
+                    (data.as_mut_ptr() as usize + data_cell_addr - vm_addr) as *mut _,
+                    RcBox::new(RefCell::new(slice::from_raw_parts_mut(
+                        data_addr as *mut u8,
+                        self.data.len(),
+                    ))),
+                );
+                data[data_addr - vm_addr..].copy_from_slice(self.data);
+            }
+
+            let region = MemoryRegion::new_writable(data.as_mut_slice(), vm_addr as u64);
+            (data, region)
+        }
+    }
+
+    #[repr(C)]
+    struct RcBox<T> {
```

Comment:
Rust std lib stability?

Reply:
I mean, we're already relying on this not changing right? The test makes it explicit at least :P